src/rust/engine/fs/store/src/local.rs

```diff
@@ -46,20 +46,27 @@ impl ByteStore {
     let directories_root = root.join("directories");
     Ok(ByteStore {
       inner: Arc::new(InnerStore {
-        // We want these stores to be allowed to grow very large, in case we are on a system with
-        // large disks which doesn't want to GC a lot.
-        // It doesn't reflect space allocated on disk, or RAM allocated (it may be reflected in
-        // VIRT but not RSS). There is no practical upper bound on this number, so we set them
-        // ridiculously high.
-        // However! We set them lower than we'd otherwise choose because sometimes we see tests on
-        // travis fail because they can't allocate virtual memory, if there are multiple Stores
-        // in memory at the same time. We don't know why they're not efficiently garbage collected
-        // by python, but they're not, so...
-        file_dbs: ShardedLmdb::new(files_root, 100 * GIGABYTES, executor.clone(), lease_time)
-          .map(Arc::new),
+        // We want these stores to be allowed to grow to a large multiple of the `StoreGCService`
+        // size target (see DEFAULT_LOCAL_STORE_GC_TARGET_BYTES), in case it is a very long time
+        // between GC runs. It doesn't reflect space allocated on disk, or RAM allocated (it may
+        // be reflected in VIRT but not RSS).
+        //
+        // However! We set them lower than we'd otherwise choose for a few reasons:
+        // 1. we see tests on travis fail because they can't allocate virtual memory if there
+        //    are too many open Stores at the same time.
+        // 2. macOS creates core dumps that apparently include MMAP'd pages, and setting this too
+        //    high will use up an unnecessary amount of disk if core dumps are enabled.
+        //
+        file_dbs: ShardedLmdb::new(
+          files_root,
+          4 * DEFAULT_LOCAL_STORE_GC_TARGET_BYTES,
```

Comment:
16 GB total means 1 GB per shard, which means we can't e.g. store any blob bigger than 1GB, right? Thinking back to when we originally set these numbers high, a hard 1GB/shard limit feels very small... In particular, I believe we assume the store is large enough to never be full (and we error, rather than force-GC, if you try to write to a full shard), so shrinking these is kinda risky...

It feels like you're performing two changes here:
1. Only mmap the per-shard limit per shard
2. Set the maximum size much lower

I think 1 is an obvious winner, but I'm not sure that 2 makes sense. Possibly _slightly_ lower (80GB?), but we picked a large number a) because much smaller numbers caused problems, and b) because a full LMDB shard is fatal... If we _do_ need to shrink this, then dropping to maybe 4 or 8 shards, so that each shard can be bigger, may make some sense.

Reply:
> 16 GB total means 1 GB per shard, which means we can't e.g. store any blob bigger than 1GB, right?

Yikes: good point.

> Possibly slightly lower (80GB?), but we picked a large number a) because much smaller numbers caused problems, and b) because a full LMDB shard is fatal

Yea, I agree. Will increase this. An 80GB core dump is much more manageable than a 1.6TB core dump in any case!